### PR TITLE
Overlapping

### PR DESCRIPTION
**Pull request by Ana06**
**Branch copied from #1106, so maybe I have to change this if #1106 changes.**

Not allow that an event is scheduled at the same time than other event in the admin schedule, as previously:

image

I've also added a validation to avoid that this happens when scheduling in different tabs. I've added tests for this validation with four cases that are representative.

### CODE REVIEW

File: app/assets/javascripts/osem-schedule.js [hunk 1]
```diff
@@ -86,8 +133,18 @@ $(document).ready( function() {
     snap: '.schedule-room-slot',
     revertDuration: 200,
     revert: function (event, ui) {
-        console.log(event.attr);
-        return !event;
+      console.log(event.attr);
```
Comment (differentreality) [inline app/assets/javascripts/osem-schedule.js]: undefined
This raises an error. Doesn't affect the PR's functionality, which seems to be working OK.

```
osem-schedule.js?body=1:136 function ( name, value ) {
		return access( this, jQuery.attr, name, value, arguments.length > 1 );
	}

```

Comment (Ana06) [inline app/assets/javascripts/osem-schedule.js]: I can remove that because it is the only log that is still here and I think it is not needed.

Comment (differentreality) [inline app/assets/javascripts/osem-schedule.js]: Sounds good

**Comment by Ana06** [comment]
As I closed #1106 and this depends on it I will close this too.

**Comment by differentreality** [comment]
Would it make sense to show on the schedule-grid only events that are confirmed? That would make schedule ignore if the events are scheduled, so that events that are scheduled and cancelled/withdrawn will be left out of the schedule grid. 
That would allow us to schedule another event in their slot. 

I am thinking something along those lines:
http://paste.opensuse.org/47699581

**Pull request closed without merging**